### ISSUE TITLE
How to test local emulator with document references

### ISSUE THREAD

**Issue by rolandjitsu** [firebase/firebase-tools#753]
### Version info

3.18.4

### Platform Information

macOS 10.12.6

### Steps to reproduce

I'm trying to use `firebase functions:shell` to test some of the functions locally.

### Expected behavior

I'm not sure how to test with an object that it's supposed to have a document reference.
Say I have a simple object with a ref:
```js
{
  timestamp: 123,
  docWithRef: DocumentReference
}
```
How can I run that in the repl that the shell created? Is that actually possible?

And I'm a little confused. When the `onCreate()` trigger that I use will invoke the callback with the data, will the snapshot actually contain `docWithRef` as a proper `DocumentReference`
 or will it be just a path?

### Actual behavior

Cannot seem to find a way to mock a doc ref.

**Comment by laurenzlong** [comment]
@rolandjitsu We don't yet support Firestore document reference or geo points in the functions shell yet. Sorry about that.

**Comment by rolandjitsu** [comment]
@laurenzlong ok, no worries. I guess I'll just test the fn after I deploy it. Are there any plans to support that in the future?

**Comment by laurenzlong** [comment]
Yes, we do plan to support it in the future. No timelines at the moment.

**Comment by lucafaggianelli** [comment]
I got this error in the emulator **The emulator does not yet support Firestore document reference values or geo points** so it appears that there's no update on this feature...any news?

**Comment by ryanpbrewster** [comment]
I think I know what is causing this issue, and it should be an easy fix.

**Comment by NoFr1ends** [comment]
Any updates on this?

**Comment by henlz** [comment]
Apparently no updates...

**Comment by bradleymackey** [comment]
Bump. With all the emulator news from the Firebase Summit '19 I'm surprised this is still outstanding.

**Comment by samtstern** [comment]
Thanks for the bump!  We will be going through the bug backlog next week to
try and smooth out all of the known emulator issues when possible.

Things got a little crazy leading up to the summit (as you can imagine) but
now we should be able to get through a lot of these.  I'll personally be
looking into this one.

**Comment by bradleymackey** [comment]
Possibly related: https://github.com/firebase/firebase-functions-test/issues/47

**Comment by samtstern** [comment]
For those waiting on progress here I have some questions:

1. Do you need support for document references in the `functions:shell`?  If so can you suggest the syntax you'd like to use for creating doc refs at the command line?
2. If you *dont* need doc ref support in the shell what sort of support do you need in the emulator?

**Comment by Dinduks** [comment]
Hi. It's been two years since this issue was opened. Any news please?

**Comment by samtstern** [comment]
We improved support for this in the emulator.  There is no update to `firebase:shell` so far.

**Comment by CambridgeAlgoRD** [comment]
When I create a document in Emulator which has got fields of type "Reference", after the creation of the document, all its fields (documentReference ones) are converted to an empty string. This is very frustrating because literally I cannot test my functions before deploying them. 
It's very strange because if I edit the field of that same document individually after creation, then I can change the field type to "Reference" with proper value successfully!

**Comment by samtstern** [comment]
@CambridgeAlgoRD that sounds like a new bug and not the central issue here (which is about `shell` support) so do you mind opening a new issue with all the steps to reproduce?

**Comment by HasarAli** [comment]
I was hoping to get a DocRef by adding a reference manually through the emulator UI, but that did not work for me either.

**Comment by yuchenshi** [comment]
@HasarAli Would you mind opening a new issue to clarify what you've tried, what is the expected behavior and the actual behavior? That will help us look more into this

**Comment by HasarAli** [comment]
@yuchenshi I just opened an [issue](https://github.com/firebase/firebase-tools-ui/issues/690#issue-1119961558)

**Comment by MrAlek** [comment]
Any update on this issue? It's been 4 years. The functions shell is great but not being able to pass document references as parameters is a huge limitation.
